crypto/siphash/siphash.c

```diff
@@ -80,12 +80,18 @@ size_t SipHash_hash_size(SIPHASH *ctx)
     return ctx->hash_size;
 }
 
-int SipHash_set_hash_size(SIPHASH *ctx, size_t hash_size)
+static size_t siphash_adjust_hash_size(size_t hash_size)
 {
     if (hash_size == 0)
         hash_size = SIPHASH_MAX_DIGEST_SIZE;
-    else if (hash_size != SIPHASH_MIN_DIGEST_SIZE &&
-             hash_size != SIPHASH_MAX_DIGEST_SIZE)
```

Comment:
hash size must be either 8 or 16. It can't be, for example, 10. But this changes makes it allowed

Reply:
There is a unit-test for size 4 to error out, as far as I know that still passes.

Reply:
`SipHash_set_hash_size` does the check and returns 0 when `hash_size` isn't one of those values, so I'm not sure what code path you're talking about.

Reply:
never mind. It's checked in another place now